[CHEC-1475] Add missing docs (#155)
chec/commerce.js: docs/02_Categories.md

products. The categories resource helps to organize products into groups. By creating categories and associating them to
your products, you can pass optional parameters to filter out the products by the `category_slug` or `category_id` in
the products endpoint. Categories data can also accessed from the products object.

---

## List categories

The `list()` method uses `GET v1/categories` to return a list of all the merchant's product categories.

Example request using Commerce.js:

```js
import Commerce from '@chec/commerce.js';

const commerce = new Commerce('{your_public_key}');

commerce.categories.list().then((category) => console.log(category.name));
```

Example request using cURL:

```bash
$ curl -X GET \
    -G "https://api.chec.io/v1/categories" \
    -H "X-Authorization: {key}"
```

| Method | Description |
| -------------------- | ----------- |
| `list(params)`       | List all categories, either by filtered params or unfiltered |

<div class="highlight highlight--note">
<span>Note</span>
  <p>Refer to the full response for listing categories <a href="/docs/api/?shell#categories">here</a>.</p>
</div>

---

## Retrieve category

> The `retrieve()` method at the `GET /v1/categories/{id}` API endpoint gets a specific category by permalink or ID.

Reviewer: ```suggestion
The `retrieve()` method uses the `GET /v1/categories/{id}` API endpoint to get a specific category by permalink or ID.
```
